pandas/formats/format.py

```diff
@@ -1951,7 +1951,11 @@ def _format_hierarchical_rows(self):
                 for spans, levels, labels in zip(level_lengths,
                                                  self.df.index.levels,
                                                  self.df.index.labels):
-                    values = levels.take(labels)
+
+                    values = levels.take(labels,
+                                         allow_fill=levels._can_hold_na,
```

Comment:
@jorisvandenbossche I'm sorry, but I can't follow you. I guess when you say

> But the problem is that the test frame only contains floats.  @mpuels can you update the test to also include an integer index case?

then you mean the MultiIndex of the DataFrame in my test case only contains floats and that I shall construct a test case where each level of the MultiIndex is of type integer?